src/net_processing.cpp

```diff
@@ -731,6 +735,37 @@ std::chrono::microseconds CalculateTxGetDataTime(const uint256& txid, std::chron
     return process_time;
 }
 
+static void RetryProcessTx(CConnman& connman, const uint256& txid, const std::chrono::microseconds current_time) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
+{
+    CNodeState::TxDownloadState* best_d = nullptr;
+    std::chrono::microseconds best;
+
+    for (auto& el : mapNodeState) {
+         CNodeState::TxDownloadState* d = &el.second.m_tx_download;
+         if (d->m_tx_in_flight.size() >= MAX_PEER_TX_IN_FLIGHT) continue;
+         auto it = d->m_tx_announced.find(txid);
+         if (it != d->m_tx_announced.end()) {
+             if (best_d == nullptr || (it->second != std::chrono::microseconds::zero() && it->second < best)) {
```

Comment:
This condition could probably be simplified considerably if (1) `best` is initialized appropriately and (2) the entry for the `txid` in `m_tx_announced` is removed on line 4079 rather than zeroed (like what is done on line 3272). Was zeroing chosen on line 4079 for a specific reason?

Reply:
My guess for the reason is [here](https://github.com/bitcoin/bitcoin/pull/18238#discussion_r397211419) :)